### PR TITLE
First attempt at CLI

### PR DESCRIPTION
**Pull request by diegostruk1**
- Adds a CLI module to interact with the pyworkflow module: making CLI a module for now, but we can consider making it its own service in the future. For now I think this might be sufficient, but this is open to discussion.
- Adds a execute_workflow which gets the order of node execution and executes it at once. This attempts to cover the: 'Provides a CLI with support for batch execution of workflows' requirement.

### CODE REVIEW

File: CLI/setup.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from setuptools import setup
+
+setup(name='CLI',
+      version='0.0.0',
+      py_modules= ['cli'],
+      install_requires=[
+            'Click',
```
Comment (reelmatt) [inline CLI/setup.py:7]: This didn't seem to install the package. Adding a line to `Pipfile` silenced the error.

File: CLI/test.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from django.test import TestCase
```
Comment (reelmatt) [inline CLI/test.py:1]: Should this be a Django TestCase or do `import unittest`?

File: CLI/test.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+from pyworkflow.pyworkflow import Workflow
```
Comment (reelmatt) [inline CLI/test.py:3]: I got a `ModuleNotFoundError`. Changing to `from pyworkflow import Workflow` worked for me.

File: CLI/test.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class WorkflowTest(TestCase):
+
+    def set_up(self):
+        self.workflow = Workflow()
+
+    def test_workflow_init(self):
+        self.assertEqual(self.workflow.name, "a-name")
```
Comment (reelmatt) [inline CLI/test.py:12]: The default Workflow name was changed to "Untitled". Might be better to pass in a name argument when calling `Workflow()` and assert based on that value.

File: README.md [hunk 5]
```diff
@@ -56,3 +56,8 @@ data from `localhost:8000` **where the Django app must be running**.
 
 ### Start dev server
 - `npm start`
+
+##CLI
+1. Run pipenv shell.
+2. Create a workflow using UI and save it. 
+3. Run it as: pyworkflow --file-directory (path-to-json-workflow-file) execute
```
Comment (reelmatt) [inline README.md:63]: I kept getting a `command not found` error when trying to the CLI as `pyworkflow`. Following the [Click guide](https://click.palletsprojects.com/en/7.x/setuptools/#testing-the-script), adding 

`pip install -editable .`

made it work. Not sure how/if this is compatible with `pipenv` (I ran the command within the `pipenv shell`), but it let me test it!

Comment (diegostruk1) [reply README.md:63]: Good call on installing that package. Will add that in the README file.

File: CLI/test.py (hunk 2 above)
Comment (diegostruk1) [reply CLI/test.py:1]: Shouldn't have added this test file. I was testing a few things but still not commit material. I will remove these.

File: CLI/test.py (hunk 4 above)
Comment (diegostruk1) [reply CLI/test.py:12]: Please see above, I unintentionally pushed this files. Will get these lines removed.

File: CLI/setup.py (hunk 1 above)
Comment (diegostruk1) [reply CLI/setup.py:7]: Added to pipfile the way Samir suggested.

File: CLI/test.py (hunk 3 above)
Comment (diegostruk1) [reply CLI/test.py:3]: Seems like both ways are working for me. However, fixing it to the way you suggest seems better.

**Pull request merged**